blockchain/chain/src/store/chain_store.rs

```diff
@@ -132,42 +136,44 @@ where
         db: DB,
         chain_config: Arc<ChainConfig>,
         genesis_block_header: &BlockHeader,
+        chain_data_root: &Path,
     ) -> Result<Self>
     where
         DB: Clone,
     {
         let (publisher, _) = broadcast::channel(SINK_CAP);
         let ts_cache = Arc::new(Mutex::new(LruCache::new(DEFAULT_TIPSET_CACHE_SIZE)));
         let genesis_ts = Arc::new(Tipset::from(genesis_block_header));
```

Comment:
Minor thing but this might be slightly shorter:
```suggestion
        let genesis_key = TipsetKeys::new(*genesis_block_header.cid()));
```